addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/handler/YeelightHandlerBase.java

```diff
@@ -200,43 +201,56 @@ public void handleCommandHelper(ChannelUID channelUID, Command command, String l
     }
 
     void handlePercentMessage(PercentType brightness) {
+        DeviceAction pAction;
         if (brightness.intValue() == 0) {
-            DeviceManager.getInstance().doAction(deviceId, DeviceAction.close);
+            pAction = DeviceAction.close;
+            pAction.putDuration(getDuration());
+            DeviceManager.getInstance().doAction(deviceId, pAction);
         } else {
             if (mDevice.getDeviceStatus().isPowerOff()) {
-                DeviceManager.getInstance().doAction(deviceId, DeviceAction.open);
+                pAction = DeviceAction.open;
+                // hard coded to fast open, the duration should apply to brightness increase only
+                pAction.putDuration(0);
+                DeviceManager.getInstance().doAction(deviceId, pAction);
             }
-            DeviceAction baction = DeviceAction.brightness;
-            baction.putValue(brightness.intValue());
-            DeviceManager.getInstance().doAction(deviceId, baction);
+            pAction = DeviceAction.brightness;
+            pAction.putValue(brightness.intValue());
+            pAction.putDuration(getDuration());
+            DeviceManager.getInstance().doAction(deviceId, pAction);
         }
     }
 
     void handleIncreaseDecreaseBrightnessCommand(IncreaseDecreaseType increaseDecrease) {
-        DeviceManager.getInstance().doAction(deviceId,
-                increaseDecrease == IncreaseDecreaseType.INCREASE ? DeviceAction.increase_bright
-                        : DeviceAction.decrease_bright);
+        DeviceAction idbAcation = increaseDecrease == IncreaseDecreaseType.INCREASE ? DeviceAction.increase_bright
+                : DeviceAction.decrease_bright;
+        idbAcation.putDuration(getDuration());
+        DeviceManager.getInstance().doAction(deviceId, idbAcation);
     }
 
     void handleIncreaseDecreaseColorTemperatureCommand(IncreaseDecreaseType increaseDecrease) {
-        DeviceManager.getInstance().doAction(deviceId,
-                increaseDecrease == IncreaseDecreaseType.INCREASE ? DeviceAction.increase_ct
-                        : DeviceAction.decrease_ct);
+        DeviceAction idctAcation = increaseDecrease == IncreaseDecreaseType.INCREASE ? DeviceAction.increase_ct
+                : DeviceAction.decrease_ct;
+        idctAcation.putDuration(getDuration());
+        DeviceManager.getInstance().doAction(deviceId, idctAcation);
     }
 
     void handleOnOffCommand(OnOffType onoff) {
-        DeviceManager.getInstance().doAction(deviceId, onoff == OnOffType.ON ? DeviceAction.open : DeviceAction.close);
+        DeviceAction ofAction = onoff == OnOffType.ON ? DeviceAction.open : DeviceAction.close;
```

Comment:
I guess, you mean `offAction`

Reply:
Ah, ok. Another example showing that the naming is not very self explaning :smile: